homeassistant/core.py

```diff
@@ -592,7 +593,7 @@ def __init__(self, domain, service, data=None, call_id=None):
         """Initialize a service call."""
         self.domain = domain.lower()
         self.service = service.lower()
-        self.data = data or {}
+        self.data = deepcopy(data) if data else {}
```

Comment:
Let's for now stick with MappingProxyType, as it won't copy the stuff. Yes, it will still allow mutating the data 1 layer deeper but at least we have 1 level of defense. We currently do the same for State attributes.